packages/fxa-settings/src/models/reliers/rellier.ts

```
@@ -0,0 +1,109 @@
+  service: string | undefined;
+  serviceName: string | undefined;
+  subscriptionProductId: string | undefined;
+  subscriptionProductName: string | undefined;
+  style: string | undefined;
+  uid: string | undefined;
+  utmCampaign: string | undefined;
+  utmContent: string | undefined;
+  utmMedium: string | undefined;
+  utmSource: string | undefined;
+  utmTerm: string | undefined;
+}
+
+/**
+ * Create a relier class that can be bound to a given context.
+ */
+export class Relier implements ContextProvider, RelierData {
+
+  @bind([V.isString, V.isRequired])
+  context: string = '';
+
+  @bind([V.isString])
+  name: string = '';
+
+  @bind([V.isString])
+  email: string | undefined;
+
+  @bind([V.isString])
+  entrypoint: string | undefined;
+
+  @bind([V.isString], T.snakeCase)
+  entrypointExperiment: string | undefined;
+
+  @bind([V.isString], T.snakeCase)
+  entrypointVariation: string | undefined;
+
+  @bind([V.isString], T.snakeCase)
+  resetPasswordConfirm: boolean | undefined;
+
+  @bind([V.isString], "setting")
+  setting: string | undefined;
+
+  @bind([V.isString], "service")
+  service: string | undefined;
+
+  // TODO: Double check
+  @bind([V.isString], "service")
+  serviceName: string | undefined;
+
+  @bind([V.isString])
+  subscriptionProductId: string | undefined;
+
+  @bind([V.isString])
+  subscriptionProductName: string | undefined;
+
+  @bind([V.isString])
+  style: string | undefined;
+
+  @bind([V.isString])
+  uid: string | undefined;
+
+  @bind([V.isString], T.snakeCase)
+  utmCampaign: string | undefined;
+
+  @bind([V.isString], T.snakeCase)
+  utmContent: string | undefined;
+
+  @bind([V.isString], T.snakeCase)
+  utmMedium: string | undefined;
+
+  @bind([V.isString], T.snakeCase)
+  utmSource: string | undefined;
+
+  @bind([V.isString], T.snakeCase)
+  utmTerm: string | undefined;
+
+  constructor(
+    protected readonly curContext:Context
+  ) { }
+
+  getContext() {
+    return this.curContext;
+  }
+
+  validate() {
```

This probably isn't needed, since the current behavior validates when data is written and read. However, we might want to change depending cost of validation. Also the first the context is loaded, it probably makes sense to validate the state and fail fast if something is off.